### ISSUE TITLE
cometvisu cannot be installed: The activate method has thrown an exception

### ISSUE THREAD

**Issue by ThKern** [openhab/openhab2-addons#3167]
Running openhabianpi, latest version 2.2.0-1.
When I try to install the cometvisu, the activation does not work, an exception is raised:

```
2018-01-24 19:16:25.638 [ERROR] [org.openhab.ui.cometvisu            ] - [org.openhab.cometvisu(272)] The activate method has thrown an exception
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Boolean
	at org.openhab.ui.cometvisu.internal.servlet.CometVisuApp.readConfiguration(CometVisuApp.java:193) ~[?:?]
	at org.openhab.ui.cometvisu.internal.servlet.CometVisuApp.activate(CometVisuApp.java:228) ~[?:?]

```
then

```
2018-01-24 19:16:25.844 [WARN ] [org.openhab.ui.cometvisu            ] - FrameworkEvent WARNING - org.openhab.ui.cometvisu
org.osgi.framework.ServiceException: org.apache.felix.scr.impl.manager.SingleComponentManager.getService() returned a null service object
	at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:232) ~[?:?] 
```

then again 

```
2018-01-24 19:16:25.851 [ERROR] [org.openhab.ui.cometvisu            ] - [org.openhab.cometvisu(272)] The activate method has thrown an exception
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Boolean
	at org.openhab.ui.cometvisu.internal.servlet.CometVisuApp.readConfiguration(CometVisuApp.java:193) ~[?:?]
	at org.openhab.ui.cometvisu.internal.servlet.CometVisuApp.activate(CometVisuApp.java:228) ~[?:?] 

```

then it is uninstalled again, at least if I install it from the CLI via

`feature:install openhab-ui-cometvisu/2.2.0
`
Installing from the UI seems to create a bigger mess, at least the automatic uninstall does not work, and I had to restart openhab.
I have also deleted the existing cometvisu.cfg, but that did not help.

Any ideas what's going wrong?
Thanks
Thomas

**Comment by ThKern** [comment]
Now after clearing all caches and several times install/uninstall via UI (not CLI) cometvisu seems to be running. However, now in the PaperUI a service is missing, the one responsible for the addon menu. in the browser console I found:

`http://openhabianpi:8080/rest httperror 500 {"error":{"message":"The template variable \u0027actionName\u0027 has no value","http-code":500,"exception":{"class":"java.lang.IllegalArgumentException","message":"The template variable \u0027actionName\u0027 has no value","localized-message":"The template variable \u0027actionName\u0027 has no value"}}}`

Restarting openhab does not help. Only uninstalling cometvisu helps.

**Comment by orificer** [comment]
I'm facing the same issue with the latest 2.3.0 snapshot running on a RasPi with openhabian (Linux openHABianPi 4.9.59-v7+ #1047).

When installing ui-cometvisu - 2.3.0.SNAPSHOT the follwing messages are shown at openhab.log:

`2018-01-31 08:26:35.427 [ERROR] [org.openhab.ui.cometvisu            ] - [org.openhab.cometvisu(254)] The activate method has thrown an exception
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Boolean
        at org.openhab.ui.cometvisu.internal.servlet.CometVisuApp.readConfiguration(CometVisuApp.java:193) ~[?:?]
        at org.openhab.ui.cometvisu.internal.servlet.CometVisuApp.activate(CometVisuApp.java:228) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
........`

`2018-01-31 08:26:35.639 [WARN ] [org.openhab.ui.cometvisu            ] - FrameworkEvent WARNING - org.openhab.ui.cometvisu
org.osgi.framework.ServiceException: org.apache.felix.scr.impl.manager.SingleComponentManager.getService() returned a null service object
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:232) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.getService(ServiceFactoryUse.java:111) ~[?:?]
........`

`2018-01-31 08:26:35.647 [ERROR] [org.openhab.ui.cometvisu            ] - [org.openhab.cometvisu(254)] The activate method has thrown an exception
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Boolean
        at org.openhab.ui.cometvisu.internal.servlet.CometVisuApp.readConfiguration(CometVisuApp.java:193) ~[?:?]
        at org.openhab.ui.cometvisu.internal.servlet.CometVisuApp.activate(CometVisuApp.java:228) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
........`

		
After the next refresh of the browser which is showing PaperUI the AddON-section is gone and it is showing an "ERROR: 500 - Internal Server Error".

![1](https://user-images.githubusercontent.com/18699518/35610506-ccb385f8-0661-11e8-8cb7-9513b57ec008.PNG)

![2](https://user-images.githubusercontent.com/18699518/35610516-d5041b0a-0661-11e8-888a-a38754e9ba15.PNG)

At the browsers console I can see the same error message as mentioned by ThKern.

**Comment by martinvw** [comment]
@peuter

**Comment by ThKern** [comment]
I think I might have nailed down the first problem (String is not bool): if in the initial config file you autodownload = 'true' then this gets converted somehow so that this true cannot be read. Maybe autodownload = B"true" is correct, but I didn't find time to try that out.
The second problem can easily be reproduced: take any empty openhab 2.2.0-1 installation, add cometvisu or cometvisu-php and you get these 500 errors. They can also be seen if you only try to open htpp://openhabianpi:8080/rest. The moment cometvisu is installed it returns an error, otherwise sensible content.

**Comment by peuter** [comment]
Yes this is a type conversion error in the code. Needs to be fixed, I'm working on it. But if you remove the autoDownload entry from the condif file this error should not occur. Can anyone of you confirm that?
So fixing this problem is easy. 

I will look for the 500 error and try to provide a fix for both problem as soon as possible.

**Comment by peuter** [comment]
Ok fixes for both problem are in PR #3197
